src/services/ethereum/transaction.ts

```diff
@@ -56,6 +59,7 @@ export class TransactionExecutor implements ITransactionExecutor {
 
     const hasCorrectNonce = await this.hasCorrectNonce(scheduled);
     if (!hasCorrectNonce.res) {
+      await this.markAllTransactions(scheduled, transactonList);
```

Comment:
Read `hasCorrectNonce` again - it returns `false` in 2 cases:
1. Nonce too high
2. Nonce too low
Only in the first case you can mark txs as stale. Otherwise it just means the txs need to wait.